### ISSUE TITLE
Effect panel accessibility tabbing

### ISSUE THREAD

**Issue by DilsonsPickles** [audacity/audacity#7836]
### Description

How the User can use their keyboard to navigate around the effects panel.

[Figma link](https://www.figma.com/design/w2kAh1gKKV5bu7htY7158W/08---Effects?node-id=4915-74712)

**Comment by DilsonsPickles** [comment]
@DavidBailes your input is most appreciated here.

**Comment by DavidBailes** [comment]
@DilsonsPickles Here are some initial thoughts:

Have you considered following these Windows guidelines for list items which contain controls:
https://learn.microsoft.com/en-us/windows/apps/design/controls/nested-ui#nested-ui-handling
- see the keyboard sub-section for the keyboard navigation.
I think its normally best to use a standard way of doing things if one is available, and also I think that tabbing to the controls in a list item is much more discoverable that using right/left arrow keys. Especially for users of screen readers, why would they know that they had to use the right/left arrow keys?
Examples of this standard behaviour: the lists of extensions in the extensions pane of visual studio code, the list of notifications in the action center in Windows 10, and the Notifications center in Windows 11,  and the list of installed apps in Windows 11 (settings>apps>Installed apps).

Concerning "Moving around the effect slot list".
I think you should be able to use the standard keystrokes for this: up/down arrow, and home/end.
I think the suggested cyclical navigation would be more confusing than useful, especially if you can move to the top and bottom of the list using home/end.

Additional actions when a list item (effect slot) is the the focus.
1. Pressing Enter opens the effect dialog (the same action as pressing the button whose label is the effect name). This would be very convenient for keyboard users, and seems a quite reasonable thing to happen.
2. A list item has a context menu which can be opened either using the mouse or keyboard. The menu would contain the commands Move up and Move down (available when applicable). For mouse users this would ensure that the reordering of the effects does not fail this WCAG criterion: https://www.w3.org/WAI/WCAG22/Understanding/dragging-movements.html. For keyboard users, it would be more discoverable than just providing keyboard shortcuts.

List item which is focused when the list gains focus (either Tab or Shift +Tab).
1. If an effect is added, then the next time the list becomes the focus, then the list item for that effect is the focus.
2. Else the list item which was the focus the last time the list was the focus (if the list has been the focus since the realtime effects pane was last opened).
3. Else the first list item.

Concerning "Switching between nested tab group items".
Especially if the change suggested above is made so that if a list item (effect slot) is the focus, then pressing Enter opens the effect's dialog, I think it would be more useful for the up/down arrow keys to move the focus to a list item (as in the standard behaviour linked to above), rather than to a corresponding control in a list item. If the latter behaviour is chosen, then care will need to be taken to avoid unnecessary verbosity for screen readers. For example, assuming the accessibility name of a list item is the name of the effect, then if the focus moves from the list item, to the power toggle button contained in it, there is no need to repeat the name of the effect. But if the focus moves from one power toggle button to another in an adjacent list item, then the name of the effect needs to be read.

**Comment by dozzzzer** [comment]
Tested

**Comment by saintmatthieu** [comment]
Estimate: 1 week. Confidence in estimate is rather low, though.

**Comment by ianalarina** [comment]
@DilsonsPickles @teetow write a list which keys user should use to reorder effects

**Comment by dozzzzer** [comment]
@grliszas14 

There's an inconsistency issue when it comes to the order of tabulation after reordering the effects.
See, if you have the effect stack open and you don't try to reorder your effects, the tabulation will always put the "X" icon in focus.
However, if you add some effects to the stack and try to reorder them using the keyboard, you will see the inconsistency. Sometimes it will be the close ("X") button, and sometimes it will be the drag handle in the topmost effect that it puts in focus when you press Tab or Shift+Tab. See how it puts the handle in focus after I press Tab at the very end.
So I assume it should always put the close button in focus instead.

https://github.com/user-attachments/assets/98b43182-ad36-48c9-89a7-1fe4bf26db8c

**Comment by chinakov** [comment]
This is done, thanks!
